Description: tests: fix zstd long option test for 32-bit architectures
Origin: upstream, https://github.com/libarchive/libarchive/commit/3bd918d92f8c34ba12de9c6604d96f9e262a59fc
Author: Martin Matuska <martin@matuska.de>
Last-Update: 2023-10-14

--- a/libarchive/test/test_write_filter_zstd.c
+++ b/libarchive/test/test_write_filter_zstd.c
@@ -161,8 +161,12 @@
 	    archive_write_set_filter_option(a, NULL, "max-frame-size", "1048576"));
 #endif
 #if ZSTD_VERSION_NUMBER >= MINVER_LONG
-	assertEqualIntA(a, ARCHIVE_OK,
-	    archive_write_set_filter_option(a, NULL, "long", "27"));
+	if ((int)(sizeof(size_t) == 4))
+		assertEqualIntA(a, ARCHIVE_OK,
+		    archive_write_set_filter_option(a, NULL, "long", "26"));
+	else
+		assertEqualIntA(a, ARCHIVE_OK,
+		    archive_write_set_filter_option(a, NULL, "long", "27"));
 	assertEqualIntA(a, ARCHIVE_FAILED,
 	    archive_write_set_filter_option(a, NULL, "long", "-1")); /* negative */
 #endif
